src/test/java/uk/gov/hmcts/reform/unspec/service/tasks/handler/StartBusinessProcessTaskHandlerTest.java

```diff
@@ -0,0 +1,135 @@
+import uk.gov.hmcts.reform.unspec.sampledata.CaseDetailsBuilder;
+import uk.gov.hmcts.reform.unspec.service.CoreCaseDataService;
+import uk.gov.hmcts.reform.unspec.service.flowstate.StateFlowEngine;
+
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.unspec.callback.CaseEvent.START_BUSINESS_PROCESS;
+
+@SpringBootTest(classes = {
+    StartBusinessProcessTaskHandler.class,
+    JacksonAutoConfiguration.class,
+    CaseDetailsConverter.class,
+    StateFlowEngine.class
+})
+@ExtendWith(SpringExtension.class)
+class StartBusinessProcessTaskHandlerTest {
+
+    private static final String CASE_ID = "1";
+    public static final String PROCESS_INSTANCE_ID = "processInstanceId";
+
+    @Mock
+    private ExternalTask mockExternalTask;
+    @Mock
+    private ExternalTaskService externalTaskService;
+    @MockBean
+    private CoreCaseDataService coreCaseDataService;
+    @Autowired
+    private StartBusinessProcessTaskHandler startBusinessProcessTaskHandler;
+
+    @BeforeEach
+    void init() {
+        when(mockExternalTask.getTopicName()).thenReturn("test");
+        when(mockExternalTask.getWorkerId()).thenReturn("worker");
+        when(mockExternalTask.getActivityId()).thenReturn("activityId");
+        when(mockExternalTask.getProcessInstanceId()).thenReturn(PROCESS_INSTANCE_ID);
+
+        when(mockExternalTask.getAllVariables())
+            .thenReturn(Map.of(
+                "caseId",
+                CASE_ID,
+                "caseEvent",
+                START_BUSINESS_PROCESS.name()
+            ));
+    }
+
+    @Test
+    void shouldUpdateBusinessStatusToStarted_whenInputStatusIsReady() {
+        CaseData caseData = new CaseDataBuilder().atStateClaimDraft()
+            .businessProcess(BusinessProcess.builder().status(BusinessProcessStatus.READY).build())
+            .build();
+
+        CaseDetails caseDetails = CaseDetailsBuilder.builder().data(caseData).build();
+
+        when(coreCaseDataService.startUpdate(eq(CASE_ID), eq(START_BUSINESS_PROCESS)))
+            .thenReturn(StartEventResponse.builder().caseDetails(caseDetails).build());
+
+        when(coreCaseDataService.submitUpdate(eq(CASE_ID), any(CaseDataContent.class)))
+            .thenReturn(caseData);
+
+        startBusinessProcessTaskHandler.execute(mockExternalTask, externalTaskService);
+
+        verify(coreCaseDataService).startUpdate(eq(CASE_ID), eq(START_BUSINESS_PROCESS));
+        verify(coreCaseDataService).submitUpdate(eq(CASE_ID), any(CaseDataContent.class));
+        verify(externalTaskService).complete(eq(mockExternalTask), any(VariableMap.class));
+    }
+
+    @Test
+    void shouldUpdateBusinessStatusToSarted_whenInputStatusIsDispatched() {
```

Comment:
These can be covered in one test using `@ParameterizedTest` and `@EnumSource(value = BusinessProcessStatus.class, names = {"READY", "DISPATCHED"})`